SHARDING.md

```diff
@@ -0,0 +1,254 @@
+# Sharding Playbook
+
+This playbook walks through the steps to create a new log shard.
+The steps should be the same for Rekor and the CT log, though the
+CT process for generating key material may differ.
+
+Roughly, the steps are:
+
+1. Create the GCP resources with Terraform
+2. Generate key material
+3. Spin up the new infrastructure
+6. Update prober
+4. Perform manual testing
+5. Update TrustedRoot with the new key material and SigningConfig with the new shard URL
+6. Turning down the old shard
+
+## Create GCP resources
+
+New shards are declared as additional modules under the respective
+environment. This is a different pattern than other modules, which are all
+initialized through the `sigstore` module. Declaring each shard as its
+own module allows us to make updates to shards one-by-one. Otherwise,
+by updating the module reference of `sigstore`, all shards would be
+concurrently updated.
+
+To add a new shard, create a module under either
+`terraform/environments/staging/1-infrastructure/staging.tf` or
+`terraform/environments/production/1-infrastructure/production.tf`.
+Name the module `rekor-tiles-log-<year>-<revision>`, e.g.
```

Comment:
what does revision mean here? That we can have multiple shards per year? Should they always count up?

nvm, somewhat answered in the example below, but some guidance on revision would still be helpful.